File: workers/workers.java
public class main{
    public static void main(String[] Args){
        Job A=new Job(800, 600, 1200);
        Job B=new Job(700, 500, 1100);
        Job C=new Job(500, 250, 1000);
        Worker a=new Worker(23, A);
        System.out.println(a.salary());
    }
}
public class Job{
    private int less, extra, exact;
    public Job(int exact, int less, int extra)
    {
        this.less=less;
        this.exact=exact;
        this.extra=extra;
    };
    public int pay(int days){
        if(days<20){
            return days*this.less;
        }else if(days==20){
            return days*this.exact;
        }else{
            return (20*exact)+(days%20)*extra;
        }
    }
}
public class Worker{
    private int days;
    private Job job;
    public Worker(int days, Job job){
        this.job=job;
        this.days=days;
    }
    public int salary(){
        return this.job.pay(this.days);
    }
}

File: workers/workers.js
class Job{
    constructor(exact, less, extra){
        this.exact=exact;
        this.extra=extra;
        this.less=less;
    }
    pay=(days)=>{
        if(days<20) return days*this.less; 
        else if(days===20) return days*this.exact;
        else return this.exact*20+this.extra*(days%20);
    }
}
class worker{
    constructor(days, job){
        this.days=days;
        this.job=job;
    }
    salary=()=> this.job.pay(this.days);
};
let A=new Job(800, 600, 1200);
let B=new Job(700, 500, 1100);
let C=new Job(500, 250, 1000);
let a=new worker(23, A);
console.log(a.salary());
